build_test/lib/src/test_bootstrap_builder.dart

```diff
@@ -76,3 +107,34 @@ class TestBootstrapBuilder extends Builder {
     }
   }
 }
+
+/// Manages a cache of [Configuration] per package, to avoid duplicating work
+/// across build steps.
+///
+/// Can safely be used across the build as configuration is invalidated by its
```

Comment:
I don't understand this line. Are you saying that test config changing causes us to throw away the asset graph?

Reply:
I am saying that this class uses the digests of the configuration files it reads to always give you an accurate result.